enclave/src/lib.rs

```diff
@@ -379,12 +385,15 @@ fn handle_shield_funds_xt(
     let account_vec = rsa3072::decrypt(&account_encrypted, &rsa_keypair)?;
     let account = AccountId::decode(&mut account_vec.as_slice()).sgx_error()?;
 
-    Stf::execute(
+    if let Err(e) = Stf::execute(
         &mut state,
         TrustedCall::balance_shield(account, amount),
         Default::default(),
         calls,
-    );
+    ) {
+        error!("Error performing Stf::execute. Error: {:?}", e);
+        return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
```

Comment:
Here I thought we let the error slide in line number 360.

Reply:
Ah i now see that this is inconsistent,with how you started to do it. Semantically, I believe it does make more sense to have `handle_call_worker_xt` and `handle_shield_funds_xt` returning errors and decide to let them slide one level above? What do you think?

Reply:
possible, but SGX_ERROR_UNEXPECTED is certainly the wrong thing to pass.

The best thing would be to really solve #82 by providing the caller (and only the caller) with the error message by buffering it for a while under a per-signer storage. But that's really out of scope for now.

But it is crucial that the enclave ecall NEVER fails because of a TrustedCall or TrustedGetter!!!

Reply:
agree that `SGX_ERROR_UNEXPECTED` is certainly the wrong thing. Changed it to return `Ok(())`